src/vsr/superblock.zig

```diff
@@ -320,23 +316,38 @@ pub const SuperBlockHeader = extern struct {
         manifest_newest_checksum_padding: u128 = 0,
         snapshots_block_checksum: u128,
         snapshots_block_checksum_padding: u128 = 0,
+        free_set_last_block_checksum: u128,
+        free_set_last_block_checksum_padding: u128 = 0,
 
         manifest_oldest_address: u64,
         manifest_newest_address: u64,
         snapshots_block_address: u64,
+        free_set_last_block_address: u64,
 
         /// The last operation committed to the state machine. At startup, replay the log hereafter.
         commit_min: u64,
 
+        // Logical storage size in bytes.
+        //
+        // If storage_size is less than the data file size, then the grid blocks beyond storage_size
+        // were used previously, but have since been freed.
+        //
+        // If storage_size is more than the data file size, then the data file might have been
+        // truncated/corrupted.
+        storage_size: u64,
+
         /// The number of manifest blocks in the manifest log.
         manifest_block_count: u32,
+        // Size of the encoded free set in bytes.
+        // It is equal to the sum of sizes of individual free set blocks and is used for assertions.
+        free_set_size: u32,
 
         // TODO Reserve some more extra space before locking in storage layout.
-        reserved: [28]u8 = [_]u8{0} ** 28,
+        reserved: [8]u8 = [_]u8{0} ** 8,
 
         comptime {
-            assert(@sizeOf(CheckpointState) == 224);
-            assert(@sizeOf(CheckpointState) % @sizeOf(u256) == 0);
+            assert(@sizeOf(CheckpointState) == 256);
+            assert(@sizeOf(CheckpointState) % @sizeOf(u128) == 0);
```

Comment:
:thinking: Stepping back, what is the purpose for this assertion? I changed it without thinking as part of the pseudo-256-bit-checksum change -- i.e., we are pretending that `manifest_oldest_checksum + manifest_oldest_checksum_padding` is a `u256`. (In the future it will be, but for the time being we don't need it and Zig doesn't support it).

Reply:
Ah, this is was unintentional change, but  I think this is correct: I think we want to align to u128, as many things are u128, and we want to align to 16 bytes, as that might help with SIMD? 

Basically, we want most things to be aligned "nicely", and 16 seems like a reasonable bound for niceness